File: backend/src/main/java/edu/miu/cs545/biddingproject/backend/domains/Customer.java
package edu.miu.cs545.biddingproject.backend.domains;

import edu.miu.cs545.biddingproject.backend.values.Name;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor @AllArgsConstructor
@Data @Builder
public class Customer {
    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String emailAddress;
    private String licenseNumber;
    @Embedded
    private Name name;
    @OneToOne(cascade = CascadeType.PERSIST)
    @JoinColumn(name = "user_id")
    private User user;
}

File: frontend/src/models/customer.js
export class Customer{

    constructor(id, firstName, lastName, email, licenseNumber){
        this.id = id;
        this.name = {
            firstName, lastName
        };
        this.emailAddress = email;
        this.licenseNumber = licenseNumber;
    }
    static createCustomer(item) {
        return new Customer(
            item.id, item.name.firstName, item.name.lastName,
            item.emailAddress, item.licenseNumber
        );
    }

}